### PR TITLE
Handle OAuth connect error responses

### PR DESCRIPTION
**Pull request by diegocurbelo**
Fixes STRIPE-1391

## Changes proposed in this Pull Request:

This PR improves errors shown when the plugin cannot start a Stripe connection.

- Request JSON explicitly for OAuth URL requests
- Show safe server error messages as plain text
- Replace missing or low-level Connect server error details with a neutral retry message
- Show a clear reload message when the admin nonce expires
- Keep the Stripe setup documentation link in inline connection error notices
- Re-enable the connect or re-authenticate button after a failed request

This PR does not change public APIs, stored options, or data formats.

## Testing instructions

1. Go to `WooCommerce → Settings → Payments → Stripe`.
2. Find an inline `Create or connect` button on the new-account screen or in the account authentication actions.
3. Open the browser console and run `wc_stripe_settings_params.oauth_nonce = 'invalid'`.
4. Click the button.
5. Confirm that the notice says: `Your session has expired. Please reload the page and try again.`
6. Confirm that the notice includes the Stripe setup documentation link and that the button is enabled again.
7. Reload the page. In the browser developer tools, block the `admin-ajax.php` request, then click the same button.
8. Confirm that the notice says: `An issue occurred generating a connection to Stripe. Please try again.`, includes the documentation link, and leaves the button enabled.
9. Stop blocking the request, reload the page, click the button, and confirm that the Stripe connection page opens.
10. If the re-authentication banner is available, set the nonce to `invalid` again and click `Re-authenticate`.
11. Confirm that the standard WooCommerce notice shows the expired-session message and that the button is enabled again. This notice does not include the documentation link.
12. Reload the page, block the `admin-ajax.php` request, and click `Re-authenticate` again.
13. Confirm that the notice says: `There was an error. Please reload the page and try again.` and that the button is enabled again.

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Tested on mobile (or does not apply)

### Changelog entry

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

File: client/settings/stripe-auth-account/connect-button.js [hunk 1]
```diff
@@ -51,35 +60,32 @@ const ConnectButton = ( {
...
-			if ( response.success && response.data.oauth_url ) {
+			if ( response?.success && response.data?.oauth_url ) {
 				window.location.assign( response.data.oauth_url );
 			} else {
-				setError( true );
-				setIsLoading( false );
-				if ( onErrorChange ) {
-					onErrorChange( true );
-				}
+				handleError( response?.data?.message );
```
Comment (wjrosa) [inline client/settings/stripe-auth-account/connect-button.js:74]: When a once expires, we fall into this path. So an admin who leaves the settings tab open long enough will see "You do not have permission to do this.". Now that this message is user-facing, `ajax_get_oauth_url` should return a "please reload the page and try again" (or something like that) for nonce failures instead of the capability error.

File: client/settings/stripe-auth-account/connection-error-notice.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ const ConnectionErrorNotice = ( { message } = {} ) => {
 				mixedString:
 					message ||
 					__(
-						'An issue occurred generating a connection to Stripe, please ensure your server has a valid SSL certificate and try again.{{br /}}For assistance, refer to our {{Link}}documentation{{/Link}}.',
+						'An issue occurred generating a connection to Stripe. Please try again.{{br /}}For assistance, refer to our {{Link}}documentation{{/Link}}.',
```
Comment (wjrosa) [inline client/settings/stripe-auth-account/connection-error-notice.js:26]: Transport-level failures (e.g. a broken CA bundle) come back from `WC_Stripe_Connect_API::request()` as a non-empty `Error: The WooCommerce Connect server returned HTTP code: 0`, so this friendly fallback never applies to exactly the retryable/SSL cases it was written for — mapping those `WP_Error` cases to admin-appropriate copy in `ajax_get_oauth_url` would fix that.

*Drafted with AI; reviewed by me.*

File: client/settings/stripe-auth-account/connection-error-notice.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ const ConnectionErrorNotice = ( { message } = {} ) => {
 				mixedString:
 					message ||
```
Comment (wjrosa) [inline client/settings/stripe-auth-account/connection-error-notice.js:24]: A server message replaces the whole `mixedString`, dropping the docs link exactly on the errors where users need help — and the raw server string is parsed as an interpolate-components template here while the reconnect banner renders the same string as plain text; appending the docs-link suffix around the message (treated as plain text) would cover both.

*Drafted with AI; reviewed by me.*

File: client/settings/stripe-auth-account/connect-button.js [hunk 4]
```diff
@@ -51,35 +60,32 @@ const ConnectButton = ( {
 			const response = await jQuery.ajax( {
 				url: ajaxurl,
 				method: 'POST',
+				dataType: 'json',
```
Comment (wjrosa) [inline client/settings/stripe-auth-account/connect-button.js:63]: This whole fetch block (ajax config, success check, redirect, message extraction) is duplicated verbatim in `re-connect-account-banner.js` — this PR had to apply the same three edits in both places, so a shared `fetchOAuthUrl( mode )` helper would prevent the next drift.

*Drafted with AI; reviewed by me.*

File: client/settings/stripe-auth-account/connect-button.js [hunk 5]
```diff
@@ -32,6 +32,15 @@ const ConnectButton = ( {
 	const buttonText = testMode
 		? __( 'Create or connect a test account', 'woocommerce-gateway-stripe' )
 		: __( 'Create or connect an account', 'woocommerce-gateway-stripe' );
+	const handleError = ( message ) => {
+		const errorValue = message || true;
```
Comment (wjrosa) [inline client/settings/stripe-auth-account/connect-button.js:36]: `message || true` makes the error channel a boolean|string union, which is why the `typeof error === 'string'` guard has to be repeated at three consumption points — always emitting the message string (or null) would delete those ternaries and prevent a future consumer from rendering `true`.

*Drafted with AI; reviewed by me.*

File: changelog.txt [hunk 6]
```diff
@@ -1,6 +1,7 @@
 *** Changelog ***
 
 xxxx-xx-xx - version 11.0.0
+* Fix - Show server error message when connect to Stripe URL could not be generated (instead of the generic SSL Cert message)
```
Comment (wjrosa) [inline changelog.txt:4]: Nit: changelog entries go at the bottom of the version block, not the top (same in readme.txt).

*Drafted with AI; reviewed by me.*

File: client/settings/stripe-auth-account/connect-button.js [hunk 7]
```diff
@@ -51,35 +60,32 @@ const ConnectButton = ( {
...
-				}
+				handleError( response?.data?.message );
 			}
 		} catch ( err ) {
-			setError( true );
-			setIsLoading( false );
-			if ( onErrorChange ) {
-				onErrorChange( true );
-			}
+			handleError( err?.responseJSON?.data?.message );
```
[the remark this answers is not in the document]
Comment (diegocurbelo) [reply client/settings/stripe-auth-account/connect-button.js:77]: Added in [00b272e](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/5850/commits/00b272eea8113fa706310718b7314fb0db4436b7)

File: client/settings/stripe-auth-account/connect-button.js (hunk 1 above)
Comment (diegocurbelo) [reply client/settings/stripe-auth-account/connect-button.js:74]: Fixed in [00b272e](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/5850/commits/00b272eea8113fa706310718b7314fb0db4436b7)

File: client/settings/stripe-auth-account/connection-error-notice.js (hunk 2 above)
Comment (diegocurbelo) [reply client/settings/stripe-auth-account/connection-error-notice.js]: Fixed in [00b272e](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/5850/commits/00b272eea8113fa706310718b7314fb0db4436b7)

File: client/settings/stripe-auth-account/connection-error-notice.js (hunk 3 above)
Comment (diegocurbelo) [reply client/settings/stripe-auth-account/connection-error-notice.js]: Addressed in [00b272e](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/5850/commits/00b272eea8113fa706310718b7314fb0db4436b7)

File: client/settings/stripe-auth-account/connect-button.js (hunk 4 above)
Comment (diegocurbelo) [reply client/settings/stripe-auth-account/connect-button.js:63]: The request shape is similar, but each caller owns different state, tracking, fallback text, and error UI. Extracting a helper would add a new return or callback contract and increase the scope of this bug fix.

Added tests to cover both paths.

We can implement this in a follow-up PR, but merging the unified onboarding for test and live modes would also change this code, so maybe that's the best place for that change,

> *Drafted with AI; reviewed by me.*

File: client/settings/stripe-auth-account/connect-button.js (hunk 5 above)
Comment (diegocurbelo) [reply client/settings/stripe-auth-account/connect-button.js]: Fixed in [00b272e](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/5850/commits/00b272eea8113fa706310718b7314fb0db4436b7)

File: includes/admin/class-wc-stripe-settings-controller.php [hunk 8]
```diff
@@ -140,7 +143,16 @@ public function ajax_get_oauth_url() {
...
 		if ( is_wp_error( $oauth_url ) ) {
-			wp_send_json_error( [ 'message' => $oauth_url->get_error_message() ] );
+			$transport_error_codes = [
+				'wcc_server_error',
+				'wcc_server_error_content_type',
+				'wcc_server_empty_response',
+			];
+			$message               = in_array( $oauth_url->get_error_code(), $transport_error_codes, true )
+				? __( 'An issue occurred generating a connection to Stripe. Please try again.', 'woocommerce-gateway-stripe' )
+				: $oauth_url->get_error_message();
```
[the remark this answers is not in the document]
Comment (diegocurbelo) [reply includes/admin/class-wc-stripe-settings-controller.php]: Addressed in [5ea7623](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/5850/commits/5ea762300b50bb06cb5e57856f22880c419343a5)

**Review by wjrosa — APPROVED** [review]
Looks good to me and works as expected! 👍  Thanks for addressing the suggestions
